### PR TITLE
102/login/authorization redirect

### PR DESCRIPTION
**Pull request by janwestphal306**
If you are not logged in you will be redirected to the sign in page.

You might want to call `db:seed` to have a user in order to use the frontend.

Tests needed to be fixed to fit the authorization. This needs to be regarded for future tests.

### CODE REVIEW

File: app/controllers/landing_controller.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
 class LandingController < ApplicationController
+  skip_before_action :authenticate_user!
```
Comment (lassesteffen) [inline app/controllers/landing_controller.rb]: Pure rails magic again :+1:

File: spec/features/sessions/user_authorization_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'Authorization', type: :feature do
+  it 'redirects the user if not logged in' do
```
Comment (lassesteffen) [inline spec/features/sessions/user_authorization_spec.rb]: Use context as describe here http://www.betterspecs.org/

The test should have two contexts 'if user is logged in' and 'if user is not logged in'. The 'it' should than only state what we expect e. g. 'it redirects the user', 'it shows a message', it should not explain the context.

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -70,7 +70,9 @@
   # arbitrary gems may also be filtered via:
   # config.filter_gems_from_backtrace("gem name")
   config.include FeatureSpecHelper, type: :feature
+  # config.include FeatureSpecHelper, type: :request
```
Comment (lassesteffen) [inline spec/rails_helper.rb]: What are you doing here?

File: spec/requests/requests_spec.rb [hunk 4]
```diff
@@ -3,10 +3,16 @@
...
+  before do
+    user = FactoryBot.create :user
+    sign_in user
+  end
+
   describe 'GET /requests' do
     it 'works! (now write some real specs)' do
-      get requests_path
-      expect(response).to have_http_status(200)
+      # get requests_path
```
Comment (lassesteffen) [inline spec/requests/requests_spec.rb]: What is the reasoning for commenting out a test? Why not remove it, if it does not make sense?

File: config/locales/devise.en.yml [hunk 5]
```diff
@@ -14,7 +14,7 @@ en:
       last_attempt: "You have one more attempt before your account is locked."
       not_found_in_database: "This user account does not exist."
       timeout: "Your session expired. Please sign in again to continue."
-      unauthenticated: "You need to sign in or sign up before continuing."
+      unauthenticated: "You need to sign in before continuing."
```
Comment (mlichtblau) [inline config/locales/devise.en.yml]: Nice

File: spec/features/sessions/user_authorization_spec.rb (hunk 2 above)
Comment (mlichtblau) [reply spec/features/sessions/user_authorization_spec.rb]: Yes good catch!

File: spec/requests/requests_spec.rb (hunk 4 above)
Comment (mlichtblau) [reply spec/requests/requests_spec.rb]: It seems to be quite the dumb test...

> (now write some real tests)

what's that supposed to mean?

File: spec/controllers/requests_controller_spec.rb [hunk 6]
```diff
@@ -26,6 +26,12 @@
 # `rails-controller-testing` gem.
 
 RSpec.describe RequestsController, type: :controller do
+  # Authenticate an user
+  before do
+    user = FactoryBot.create :user
+    sign_in user
```
Comment (mlichtblau) [inline spec/controllers/requests_controller_spec.rb]: If you want to save a line you could write `sign_in FactoryBot.create(:user)` as you don't actually use the user variable anywhere else.

**Comment by lassesteffen** [comment]
Seems good now, would be cool if you can update the branch and than merge this.

File: spec/features/sessions/user_authorization_spec.rb (hunk 2 above)
Comment (janwestphal306) [reply spec/features/sessions/user_authorization_spec.rb]: Thanks. It really looks better that way.

File: spec/requests/requests_spec.rb (hunk 4 above)
Comment (janwestphal306) [reply spec/requests/requests_spec.rb]: I did not see this in the commit. The test still exists but is now uncommented.

File: spec/controllers/requests_controller_spec.rb (hunk 6 above)
Comment (janwestphal306) [reply spec/controllers/requests_controller_spec.rb]: I implemented this your way. If some test need the user, it can always become a variable.

**Pull request merged**